modules/contact/test/tests/mortar_tm/2d/frictionless_first/finite_noaction.i

```diff
@@ -201,6 +201,7 @@ name = 'finite_noaction'
   dtmin = 0.1
   timestep_tolerance = 1e-6
   line_search = 'contact'
+  snesmf_reuse_base = false
```

Comment:
I have not see much of a difference flipping this Boolean recently: Does using dual basis functions "cancel" its effect?

Reply:
If you are using our new mortar objects, you should not see any effect because they are not stateful (thank goodness), e.g. F(u) is always the same for a given u. The old non-weighted-gap way seems to have potentially been stateful, or some other object in these inputs were stateful such that F(u) is not always the same for a given u.

Reply:
Got it

Reply:
Maybe there's already logic, but can you throw a paramError if this is set and using the new mortar objects? I have a big personal distaste for input parameters that don't always do something since some one much less capable (me) has to track down what they do.

Reply:
What do you mean by "don't always do something"? This parameter will always do something. It toggles on/off whether we reuse the nonlinear residual as the base linear residual for matrix-free methods. This defaults to true in MOOSE but users can set it to false if they have to...and most contact methods have to, just not this new mortar. Auto-toggling of this parameter is pretty easy to achieve from an action (it's automatically toggled to `false` in the `ContactAction`) but it would be somewhat ugly to try to pull the `Executioner` option into the mortar constraint...then you would be getting a `paramError` from a block (the constraint block) that is not logically associated with the block you are issuing a parameter error about